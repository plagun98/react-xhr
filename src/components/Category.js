import React from 'react';
import { Row } from 'react-bootstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import CardItem from './Card';


const Category = props => {
    return (
        <Row>
            {
                props.categoryStorage.items.map(product => {
                    if(product.bsr_category === props.category) {
                        let str = props.prop.location.search.slice(1).toLowerCase();
                        if(~product.name.toLowerCase().indexOf(str)){
                            return <CardItem key={product.asin} name={product.name} brand={product.brand} price={product.price} currency={product.currency} asin={product.asin} weight={product.weight} img={product.img} link={product.link}/>
                        }
                    }
                    return null;
                })
            }
        </Row>
    );
} 
Category.propTypes = {
    categoryStorage: PropTypes.shape({
        searchData: PropTypes.string,
        category: PropTypes.string,
        items: PropTypes.arrayOf(
            PropTypes.object
        )
    })
}

export default connect(
    state => ({
        categoryStorage: state
    })
)(Category);