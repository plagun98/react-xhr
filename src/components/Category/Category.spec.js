import Category from './Category';

it('render test', () => {
    expect(Category).toMatchSnapshot()
});