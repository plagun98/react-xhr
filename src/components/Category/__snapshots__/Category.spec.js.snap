// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `
Object {
  "$$typeof": Symbol(react.memo),
  "WrappedComponent": [Function],
  "compare": null,
  "displayName": "Connect(Category)",
  "type": [Function],
}
`;
